plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -31,11 +42,23 @@ export class AzureIdentityKubernetesAuthTranslator
       clusterDetails,
     );
 
-    const credentials = new DefaultAzureCredential();
+    if (!this.accessToken || this.tokenExpired()) {
+      this.accessToken = await this.tokenCredential.getToken(aksScope);
+
+      if (!this.accessToken) {
+        throw new Error('Unable to retrieve Azure token');
+      }
+    }
 
-    // TODO: can we cache this? It's inneficiant to get a new token every time
-    const accessToken = await credentials.getToken(aksScope);
-    clusterDetailsWithAuthToken.serviceAccountToken = accessToken.token;
+    clusterDetailsWithAuthToken.serviceAccountToken = this.accessToken.token;
     return clusterDetailsWithAuthToken;
   }
+
+  private tokenExpired(): boolean {
+    if (!this.accessToken) return true;
+
+    // Set tokens to expire 2 minutes before its actual expiry time
```

Comment:
I think 2 minutes is perhaps a bit short, maybe at least 10 minutes? or even 30.

Reply:
The goal was to avoid using a token that will expire mid-flight, so that's why I thought 2min would be enough. I changed to 15 minutes, what you think? I think 30 is way too long and reduces some of the benefit from caching it.